### PR TITLE
Add pronoun framework

### PR DESCRIPTION
**Pull request by tazz4843**
This PR aims to add pronouns to users. It adds a new enum named Pronouns and adds the field as optional to users and user update/create JSON.

### CODE REVIEW

**Comment by hydrostaticcog** [comment]
maybe just me, but i don't intend on setting pronouns. Not that I don't like them or have anything against them, I just saw think it's unnecessary. Maybe adding a null value could also be helpful for a default as well.

**Comment by tazz4843** [comment]
they will be null by default

**Comment by hydrostaticcog** [comment]
fantastic

File: src/types/pronouns.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            10 => SheIt,
+            11 => SheThey,
+            12 => TheyHe,
+            13 => TheyIt,
+            14 => TheyShe,
+            15 => TheyThem,
+            16 => Any,
+            17 => OtherAsk,
+            18 => Avoid,
+            _ => panic!("got {}, out of range for pronoun conversion", x),
```
Comment (randomairborne) [inline src/types/pronouns.rs:69]: this panic can never happen outside of server startup, is that correct?

**Review by randomairborne — CHANGES_REQUESTED** [review]

File: src/types/pronouns.rs (hunk 1 above)
Comment (tazz4843) [reply src/types/pronouns.rs:69]: this panic can happen anywhere, but there's no way for a value out of range to be set in the database, making it essentially impossible for this to be reached

Comment (randomairborne) [reply src/types/pronouns.rs:69]: still don't panic, handle the error, just in case

**Review by randomairborne — APPROVED** [review]
Looks good, nowhere that could crash the server

**Pull request merged**